src/gui/updater/ocupdater.cpp

```diff
@@ -104,15 +104,23 @@ bool OCUpdater::performUpdate()
     QString updateFile = settings.value(updateAvailableC).toString();
     if (!updateFile.isEmpty() && QFile(updateFile).exists()
         && !updateSucceeded() /* Someone might have run the updater manually between restarts */) {
-        const QString name = Theme::instance()->appNameGUI();
-        if (QMessageBox::information(nullptr, tr("New %1 Update Ready").arg(name),
-                tr("A new update for %1 is about to be installed. The updater may ask\n"
-                   "for additional privileges during the process.")
-                    .arg(name),
-                QMessageBox::Ok)) {
+        const auto messageBoxStartInstaller = new QMessageBox(QMessageBox::Information,
+            tr("New %1 Update Ready").arg(Theme::instance()->appNameGUI()),
+            tr("A new update for %1 is about to be installed. The updater may ask\n"
+               "for additional privileges during the process.")
+                .arg(Theme::instance()->appNameGUI()),
+            QMessageBox::Ok,
+            nullptr);
+
+        messageBoxStartInstaller->setAttribute(Qt::WA_DeleteOnClose);
+
+        connect(messageBoxStartInstaller, &QMessageBox::accepted, this, [this] {
             slotStartInstaller();
-            return true;
-        }
+        });
+        connect(messageBoxStartInstaller, &QMessageBox::rejected, this, [this] {
+            slotStartInstaller();
```

Comment:
I am confused about this one, is this not a behavior change from before ?
why do you need it ?

Reply:
@mgallien Yes. I need to change behavior. Otherwise, I can not fix the issue, because, it is caused by incorrect behavior. We must not return true. Otherwise, `QCoreApplication` would never enter the event loop, because, this code:
```
if (updater && updater->handleStartup()) {
    return 1;
}
```
in `src/gui/main.cpp`.

That's why, we must not return true when displaying the message box. Also, we should be able to call `qApp->quit()` when closing this message box. Otherwise, if we call it right away when displaying the message box, the `QCoreApplication` hasn't entered the event loop yet. This leads to calling `qApp->quit()` having no effect because the event loop hasn't started yet. According to Qt documentation.

Reply:
in that case, you may use QDialog::finished such that you avoid needing to connect to two signals

Reply:
@mgallien At first, I wanted to use it, but then, I've decided to make this code more deterministic so I only execute that line on accepting or rejecting and no other reason for dialog finish.

Will that `QDialog::finished` be triggered only when I close the dialog via the X button (ESC on keyboard) and via OK button (ENTER on keyboard) ? If that is the case, I am going to use `QDialog::finished`.

Reply:
`finished` should happen only for `accepted` or `rejected` in our case
it is always possible to call the `done` slot with another return code but the user can not cause that by closing the dialog
you should be safe using it